docs/tutorials/05_kv-scheduler.md

```diff
@@ -0,0 +1,303 @@
+```
+
+* Northbound key prefix for configuration type handled by the descriptor
+```go
+NBKeyPrefix: "/interface/",
+```
+
+* String representation of the type
+```go
+ValueTypeName: proto.MessageName(&model.Interface{}),
+```
+
+* Configuration item identifier (label, name, index) is returned by this method. 
+```go
+KeyLabel: func(key string) string {
+    return strings.TrimPrefix(key, "/interface/")
+},
+```
+
+* Key selector returns true if the provided key is valid for given descriptor. In the example, every descriptor uses own key, but there are cases where the descriptor processes other descriptor's keys or derived types.
+```go
+KeySelector: func(key string) bool {
+    return true	
+},
+```
+
+* This flag enables metadata for given type
+```go
+WithMetadata: true,
+```
+
+* Create method configures new configuration item.
+```go
+Create: func(key string, value *model.Car) (metadata interface{}, err error) {
+    d.log.Infof("%s car created", value.Color)
+    // Return color of the car so the scheduler remembers it
+    return value.Color, nil
+},
+```
+
+This is how the completed interface descriptor looks like:
+```go
+return &adapter.CarDescriptor{
+    Name: "interface-descriptor",
+    NBKeyPrefix: "/interface/",
+    ValueTypeName: proto.MessageName(&model.Interface{}),
+    KeyLabel: func(key string) string {
+        return strings.TrimPrefix(key, "/interface/")
+    },
+    KeySelector: func(key string) bool {
+        return true	
+    },
+    WithMetadata: true,
+    Create: func(key string, value *model.Interface) (metadata interface{}, err error) {
+        d.log.Infof("Interface %s created", value.Name)
+        return value.Color, nil
+    },
+}
+```
+
+#### 3. Descriptor with dependency
+
+This descriptor defines some additional fields, since we need to define dependency on the interface configuration item. Define the constructor and related methods first:
+```go
+type RouteDescriptor struct {
+	// dependencies
+	log logging.PluginLogger
+}
+
+func NewRouteDescriptor(logger logging.PluginLogger) *RouteDescriptor {
+	return &RouteDescriptor{
+		log: logger,
+	}
+}
+
+func (d *RouteDescriptor) GetDescriptor() *adapter.RouteDescriptor {
+	return &adapter.RouteDescriptor{}
+	}
+}
+```
+
+The route descriptor fields `Name`, `NBKeyPrefix`, `ValueTypeName`, `KeyLabel`and `KeySelector` are implemented in the same manner as for the interface type. The field `WithMetadata` is not needed here, so the `Create` method does not return any metadata value:
+```go
+Create: func(key string, value *model.Route) (metadata interface{}, err error) {
+    d.log.Infof("Created route %s dependent on interface %s", value.Name, value.InterfaceName)
+    return nil, nil
+},
+```
+
+In addition, there are two new fields:
+
+* A list of dependencies - complete keys (prefix+label) required for the given configuration item. The item will not be created while the dependent key does not exist. The label is informative and should be unique.
```

Comment:
"complete keys (prefix+label)" - actually, the keys are not absolutely complete, the agent prefix is not used inside the scheduler